app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -99,6 +99,7 @@ def index_relation_product(query)
     def index_relation_content_view(query)
       if params[:content_view_version_id]
         query = query.where(:content_view_version_id => params[:content_view_version_id])
+        query = query.archived unless params[:library] || params[:environment_id]
```

Comment:
should we update the filter_by_content_view() method to also take archived into account?  For example if the user specifies content_view_id and environment_id